src/Microsoft.Diagnostics.Monitoring.RestServer/ProcessFilterOptionsUtils.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.RestServer
+{
+    internal static class ProcessFilterOptionsUtils
+    {
+        public static bool MatchFilter(ProcessFilterDescriptor filter, IProcessInfo processInfo)
+        {
+            if (filter.Key == ProcessFilterKey.Pid)
+            {
+                return ExactCompare(filter, processInfo.EndpointInfo.ProcessId.ToString(CultureInfo.InvariantCulture));
+            }
+            if (filter.Key == ProcessFilterKey.CommandLine)
+            {
+                return Compare(filter, processInfo.CommandLine);
+            }
+            if (filter.Key == ProcessFilterKey.Name)
+            {
+                return Compare(filter, processInfo.ProcessName);
+            }
+            if ((filter.Key == ProcessFilterKey.RuntimeId) && (processInfo.EndpointInfo.RuntimeInstanceCookie != Guid.Empty))
+            {
+                return ExactCompare(filter, processInfo.EndpointInfo.RuntimeInstanceCookie.ToString("D"));
+            }
+
```

Comment:
Can you Debug.Assert/Fail here so that if we add another filter criteria, it's easier to find this if a mistake was made.